src/sentry/api/endpoints/project_servicehook_details.py

```diff
@@ -67,12 +68,16 @@ def put(self, request, project, hook_id):
         result = validator.object
```

Comment:
How does this endpoint work with OAuth? Is `authentication_classes` and `permission_classes` not being defined mean this won't be accessible via oauth apps?

Reply:
@mnoble by default we support "all forms of authentication"

```python
    authentication_classes = DEFAULT_AUTHENTICATION
```

```python
DEFAULT_AUTHENTICATION = (
    TokenAuthentication, ApiKeyAuthentication, SessionAuthentication, )
```

Permission defaults to none, but we override that with the base classes.

```python
    permission_classes = (NoPermission, )
```

becomes

```python
    permission_classes = (ProjectPermission, )
```

Reply:
Got it 👍